### ISSUE TITLE
Android: Use documents folder instead of App specific files

### ISSUE THREAD

**Issue by DominusExult** [exult/exult#604]
see https://developer.android.com/training/data-storage

> Android uses a file system that's similar to disk-based file systems on other platforms. The system provides several options for you to save your app data:
> 
> - App-specific storage: Store files that are meant for your app's use only, either in dedicated directories within an internal storage volume or different dedicated directories within external storage. Use the directories within internal storage to save sensitive information that other apps shouldn't access.
> - Shared storage: Store files that your app intends to share with other apps, including media, documents, and other files.
> - Preferences: Store private, primitive data in key-value pairs.
> - Databases: Store structured data in a private database using the Room persistence library.

**Shared storage** would be the documents folder for example.

To me it would make sense to at least use the documents folder for our savegames. Uninstalling Exult on Android deletes **ALL** files right now, including the savegames.
And even using documents for *everything* (game files, mods, data, savegames) would have good reasons, IMO.
AFAIK, accessing documents with some file transfer applications from desktop OS' is possible so this gives you more freedom to access the files again.

However, I have no idea if that is as easy as I'm thinking it is.

**Comment by DominusExult** [comment]
Reading more on this, and testing on Android (generally not related to this), I have to rework that feature request. Adding everything to documents is a no go. Savegames perhaps even though that gets frowned upon, too.
The problem is that it might clutter the view in the files app, as it seems to me that it by default is an unorganized view of every file on the device that is available :)

**Comment by lje** [comment]
[ScummVM](https://github.com/scummvm/scummvm) currently uses [SAF](https://developer.android.com/guide/topics/providers/document-provider) to access game files on Android. SAF is slow if a lot of files are involved, which leads to long loading times for some games like Gabriel Knight. Loading times should be tested if you intend to go that route with Exult (for game files, just using it for save games probably works well).

**Comment by i30817** [comment]
Exult or rather the android launcher (can) use SAF to copy files to the app data dir. 

I doubt they want to change this. 

Btw, SAF is very useful in scummvm to allow using remote servers for the game location (with RSAF). At the cost of EVEN more slowness, for the larger games.

**Comment by DominusExult** [comment]
Exult and the launcher use internal/app specific storage.
I would very much like us to use SAF for savegames.
Internal storage is fine for stuff that can easily be installed through the launcher. But savegames are getting lost when you reinstall Exult and that's not a good solution ;(

**Comment by i30817** [comment]
I literally used the launcher to connect to a computer in the local network to copy the files (using RSAF, which is a SAF plugin).

Just opening the system filechooser to copy the files gives you a kind of "soft SAF" because the system filechooser can use SAF plugins to access places you'd not be expected to.

Sure, afterwards you're using app storage, but the important part for exult is the first, because, frankly, the game files and mods for exult are less than 200 mb.

**Comment by i30817** [comment]
Actually the biggest annoyance of this is actually that all games and mods and bios... everything, needs to be in a zip. That's a bit annoying, if you're trying to do that remotely. I had to ssh to massage the data.

**Comment by i30817** [comment]
Also, clicking the checkbox to skip the launcher means you can't get it back to add more things without deleting all the data already installed as far as I know. Very annoying, maybe a context launcher in the exult menu like I saw other apps do with a long press of the icon? RSAF does this.

**Comment by wench** [comment]
The skipping launcher thing is really annoying. If I accidentally press it, I end up using an adb shell to get the launcher back not something to advise to users. To transfer files I have used netcat from a shell

For the launcher I was thinking that when you choose to exit exult it should always return to the launcher effectively unckecking the skip checkbox if it was checked. I think I will implement this change 

I was thing th

**Comment by wench** [comment]
I think I wil also l add a setting to the ingame options grumps to toggle the Android launcher.

**Comment by i30817** [comment]
I would think you'd prefer to use the long touch in the icon thing, the exult options are really crowded.

**Comment by wench** [comment]
Yeah but I don't know how to implement long touch  thing and having anyway to fix it is better than nothing and I can do this.

**Comment by i30817** [comment]
https://developer.android.com/develop/ui/views/launch/shortcuts

It's a intent xml thing in the android package.

**Comment by wench** [comment]
Ok I'll see what I can do

**Comment by wench** [comment]
I have the shortcut thing figured out and will commit this stuff tomorrow

**Comment by wench** [comment]
Pushed my Launcher related changes.
Added "Force Start Launcher" shortcut accessed with Long Press. Feel free to edit strings.xml  if anyone wants to change the shortcut label.
also added a setting to "Game Display Options Gump" to change Android Autolaunch from within Exult proper.

**Comment by DominusExult** [comment]
works nicely! Thank you
